core/server/common/src/main/java/alluxio/master/journal/AsyncJournalWriter.java

```diff
@@ -39,17 +79,39 @@
   private final AtomicLong mFlushCounter;
   /**
    * Represents the count of entries written to the journal writer.
+   * This counter is only accessed by the dedicated journal thread.
    * Invariant: {@code mWriteCounter >= mFlushCounter}
    */
-  private final AtomicLong mWriteCounter;
+  private Long mWriteCounter;
   /** Maximum number of nanoseconds for a batch flush. */
   private final long mFlushBatchTimeNs;
 
   /**
-   * Use a {@link ReentrantLock} to guard the journal writing. Using the fairness policy seems to
-   * result in better throughput.
+   * List of flush tickets submitted by ::flush() method.
    */
-  private final ReentrantLock mFlushLock = new ReentrantLock(true);
+  @GuardedBy("mTicketLock")
+  private final List<FlushTicket> mTicketList = new ArrayList<>(200);
+
+  /**
+   * Used to guard access to ticket cache.
+   */
+  private final ReentrantLock mTicketLock = new ReentrantLock(true);
```

Comment:
For what? Getting rid of semaphore?

Reply:
Instead of using a `ReentrantLock`, we could create an `Object` and use `synchronized(mTicketLock)` since it doesn't seem like we are leveraging any of the functionality provided by reentrant lock

Reply:
I see. done.